src/js/videojs-dash.js

```diff
@@ -261,7 +273,7 @@ class Html5DashJS {
   }
 
   duration() {
-    if (this.mediaPlayer_.isDynamic()) {
+    if (this.mediaPlayer_.isDynamic() && !this.hasFiniteDuration_) {
```

Comment:
Is it possible for us to not update `this.hasFiniteDuration_` before `duration` is requested and thus we'll be stuck in live mode?

Reply:
I don't remember exactly where it is in video.js, but the thing that makes it live is a duration check in the control bar, right? If it's listening to `durationchange`, then we could fire one off when the manifest loads (i.e. in `getDuration`) to make sure the control bar always updates to be live/not live. That should cover any situation where we're streaming a live feed that becomes non-live.

Reply:
Ah yes, seems to be [here](https://github.com/videojs/video.js/blob/master/src/js/control-bar/live-display.js#L29). Should I add in that `trigger`?

Reply:
seems reasonable to add a durationchange trigger.